pkg/reconciler/events/cloudevent/cloudeventsfakeclient.go

```diff
@@ -72,7 +79,69 @@ func (c FakeClient) StartReceiver(ctx context.Context, fn interface{}) error {
 	return nil
 }
 
+// AddCount increases 1 count for waitGroup
+func (c FakeClient) addCount() {
+	c.waitGroup.Add(1)
+}
+
+// DecreaseCount decreases 1 count for waitGroup
+func (c FakeClient) decreaseCount() {
+	c.waitGroup.Done()
+}
+
```

Comment:
I wonder if `decreaseCount` is even necessary? you could call c.waitGroup.Done() within`Send`

Reply:
Probably not? Because the fakeclient `Send` is not only used in `SendCloudEventWithRetries`'s gorountine. There is another function `SendCloudEvents` also used it and it doesn't need the waitgroup to block.

Reply:
I'm not sure since that's confusing. 😂 We don't have gorontine in `SendCloudEvents `